### DISCUSSION TITLE
[Enhancement] [date-picker] 日期选择器支持周和年范围选择

### DISCUSSION THREAD

**Discussion by lyric-zemin** [element-plus/element-plus#16816] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `date-picker`

## Description

目前日期选择器支持`datetimerange | daterange | monthrange`三种范围选择，希望增加周和年的范围选择

### Additional comments

_(empty)_

![image](https://github.com/element-plus/element-plus/assets/54665054/68adfdd8-0322-4211-9aaf-5b410f1ee2c6)

**Comment by warmthsea** [comment]
欢迎 PR

**Reply by chenlongkuan** [reply]
还有季度的范围选择，也希望加上，拜托~

**Comment by lyric-zemin** [comment]
@warmthsea 能review下吗 [link](https://github.com/element-plus/element-plus/pull/16896)
